packages/@lwc/integration-karma/test/rendering/callback-invocation-order/index.spec.js

```diff
@@ -228,3 +229,54 @@ it('should invoke callbacks on the right order (issue #1199 and #1198)', () => {
         );
     });
 });
+
+it('should invoke callbacks on the right order when multiple templates are used with lwc:if', () => {
```

Comment:
I feel like we went through a similar exercise with scoped slots and VFragments, but here we have to decide on what the "right order" is.

This one is trickier because not only is the LWC runtime's general ordering inconsistent with native (https://github.com/salesforce/lwc/issues/2713), but it also seems internally inconsistent; between `lwc:if` and `if:true`, native and synthetic modes, 4 different functions trigger the disconnectedCallback directly or indirectly, using both orderings.

This test exercises two different methods of unrendering/rerendering, which exercise 2 different parts of the code in both synthetic and native shadow modes:
1. Swapping a template for another one in a custom render function.
2. Toggling a value to hide content wrapped inside a `if:true`/`lwc:if`.